### PR TITLE
Add BaseVideoProvider::getReferenceUrl() method

### PR DESCRIPTION
**Pull request by EmmanuelVella**
This PR add a ``BaseVideoProvider::getReferenceUrl()`` public abstract method, just to easily get the original URL.

### CODE REVIEW

**Comment by core23** [comment]
Are you still working on this? Please squash and rebase or close this PR @EmmanuelVella

File: Tests/Provider/DailyMotionProviderTest.php [hunk 1]
```diff
@@ -187,4 +187,12 @@ public function testHelperProperies()
         $this->assertEquals(100, $properties['height']);
         $this->assertEquals(100, $properties['width']);
     }
+
+    public function testGetReferenceUrl()
+    {
+        $media = new Media;
```
Comment (OskarStark) [inline Tests/Provider/DailyMotionProviderTest.php]: please write `new Media()` instead of `new Media`

File: Tests/Provider/VimeoProviderTest.php [hunk 2]
```diff
@@ -189,4 +189,12 @@ public function testHelperProperies()
         $this->assertEquals(100, $properties['width']);
 
     }
+
+    public function testGetReferenceUrl()
+    {
+        $media = new Media;
```
Comment (OskarStark) [inline Tests/Provider/VimeoProviderTest.php]: please write `new Media()` instead of `new Media`

File: Tests/Provider/YouTubeProviderTest.php [hunk 3]
```diff
@@ -186,4 +186,12 @@ public function testHelperProperties()
         $this->assertEquals(100, $properties['player_parameters']['height']);
         $this->assertEquals(100, $properties['player_parameters']['width']);
     }
+
+    public function testGetReferenceUrl()
+    {
+        $media = new Media;
```
Comment (OskarStark) [inline Tests/Provider/YouTubeProviderTest.php]: please write `new Media()` instead of `new Media`

File: Provider/BaseVideoProvider.php [hunk 4]
```diff
@@ -158,6 +158,15 @@ public function postRemove(MediaInterface $media)
     }
 
     /**
+     * Get provider reference url
+     *
+     * @param \Sonata\MediaBundle\Model\MediaInterface $media
```
Comment (core23) [inline Provider/BaseVideoProvider.php]: No FQN required

**Comment by OskarStark** [comment]
@EmmanuelVella can you please apply our small comments?

after that we can merge this PR

**Comment by EmmanuelVella** [comment]
Hey, sorry I didn't see the github notifications. I just made the changes !

**Pull request closed without merging**